pdns/dns_random.cc

```diff
@@ -210,7 +210,7 @@ void dns_random_init(const string& data __attribute__((unused)), bool force) {
 }
 
 /* Parts of this code come from arc4random_uniform */
-unsigned int dns_random(unsigned int upper_bound) {
+uint32_t dns_random(uint32_t upper_bound) {
```

Comment:
It would be nice to fix the return type for the `getrandom()` and `openssl` cases as well.

Reply:
Also `ULONG_MAX` should probably be changed to `UINT32_MAX`.